### PR TITLE
 Registration Inline Hook guide single-page conversion (#2791) from Staging

### PR DESCRIPTION
**Pull request by brianduffield-okta**
## Description:
- **What's changed?** Converting Registration Inline Hook Guide into single-page guide
- **Is this PR related to a Monolith release?** n/a

### Resolves:

* [OKTA-438766](https://oktainc.atlassian.net/browse/OKTA-438766)

### CODE REVIEW

File: packages/@okta/vuepress-site/docs/guides/registration-inline-hook/main/index.md [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+* Preview and test a Registration Inline Hook
+
+**What you need**
+
+* A [Glitch.com](https://glitch.com) project or account.
+* An Okta developer org. [Create an org for free](https://developer.okta.com/signup/).
+
+**Sample code**
+
+* [Okta Registration Inline Hook Example](https://glitch.com/~okta-inlinehook-registrationhook)
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/guides/registration-inline-hook/main/index.md:24]:
```suggestion
[Okta Registration Inline Hook Example](https://glitch.com/~okta-inlinehook-registrationhook)
```

File: packages/@okta/vuepress-site/docs/guides/registration-inline-hook/main/index.md [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+
+* [Event Hook](/docs/guides/event-hook-implementation/)
+* [Password Import Inline Hook](/docs/guides/password-import-inline-hook/)
+* [Token Inline Hook](/docs/guides/token-inline-hook/)
+
+## See also
+
+For further background on the Registration Inline Hook, see:
+
+* [Registration Inline Hook](/docs/reference/registration-hook/) for a complete description of this Inline Hook type.
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/guides/registration-inline-hook/main/index.md:155]: Might just make the whole thing one sentence: For further background on the Registration Inline Hook, see [Registration......]().....

```suggestion
[Registration Inline Hook](/docs/reference/registration-hook/) for a complete description of this Inline Hook type.
```

File: packages/@okta/vuepress-site/docs/guides/registration-inline-hook/main/index.md [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+
+---
+
+In the following example, the external service code parses requests from Okta and responds with commands that indicate whether the end user's email domain is valid and allowed to register.
+
+At a high-level, the following workflow occurs:
+
+1. A user attempts to self-register for your Okta org.
+1. A Registration Inline Hook fires during this process and sends a call to the external service with the user's data.
+1. The external service evaluates the Okta call to make sure the user is from domain "example.com".
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/guides/registration-inline-hook/main/index.md:34]:
```suggestion
1. The external service evaluates the Okta call to make sure the user is from domain `example.com`.
```

File: packages/@okta/vuepress-site/docs/guides/registration-inline-hook/main/index.md [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+
+## Add request code
+
+This step includes the code that parses the body of the request received from Okta, which gets the values of `data.userProfile`. These properties contain the credentials submitted by the end user who is trying to self register.
+
+<StackSelector snippet="get-submitted-credentials"/>
+
+## Send response
+
+The external service responds to Okta indicating whether to accept the user self-registration, or not, by returning a `commands` object in the body of the HTTPS response, using a specified syntax within the object to indicate to Okta that the user should either be denied or allowed to self-register.
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/guides/registration-inline-hook/main/index.md:45]:
```suggestion
The external service responds to Okta indicating whether to accept the user self-registration by returning a `commands` object in the body of the HTTPS response, using a specified syntax within the object to indicate to Okta that the user should either be denied or allowed to self-register.
```

File: packages/@okta/vuepress-site/docs/guides/registration-inline-hook/main/index.md [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+
+The Registration Inline Hook must be set up and activated and enabled within your Okta Admin Console.
+
+To set up and activate the Registration Inline Hook:
+
+1. In the Admin Console, go to **Workflow** > **Inline Hooks**.
+2. Click **Add Inline Hook** and select **Registration** from the drop-down menu.
+3. Add a name for the hook (in this example, "Guide Registration Hook Code").
+4. Add your external service URL, including the endpoint. For example, use your Glitch project name with the endpoint:  `https://your-glitch-projectname.glitch.me/registrationHook`.
+5. Include authentication field and secret. In this example:
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/guides/registration-inline-hook/main/index.md:59]:
```suggestion
5. Include the authentication field and secret. In this example:
```

File: packages/@okta/vuepress-site/docs/guides/registration-inline-hook/main/index.md [hunk 6]
```diff
@@ -0,0 +1,155 @@
...
+
+    * **Authentication Field** = `authorization`
+    * **Authorization Secret** = `Basic YWRtaW46c3VwZXJzZWNyZXQ=`
+6. Click **Save**.
+
+The Registration Inline Hook is now set up with a status of active.
+
+> **Note:** You can also set up an inline hook using an API. See [Inline Hooks Management API](/docs/reference/api/inline-hooks/#create-inline-hook) for further information.
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/guides/registration-inline-hook/main/index.md:67]:
```suggestion
> **Note:** You can also set up an inline hook using an API. See [Inline Hooks Management API](/docs/reference/api/inline-hooks/#create-inline-hook).
```

File: packages/@okta/vuepress-site/docs/guides/registration-inline-hook/main/index.md [hunk 7]
```diff
@@ -0,0 +1,155 @@
...
+
+### Enable the Registration Inline Hook
+
+The procedure to enable the Registration Inline Hook is dependent on the type of org you are using: Okta Identity Engine or Okta Classic Engine. Follow the procedure for your specific org.
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/guides/registration-inline-hook/main/index.md:71]:
```suggestion
The procedure to enable the Registration Inline Hook is dependent on the type of org that you are using: Okta Identity Engine or Okta Classic Engine. Follow the procedure for your specific org.
```

File: packages/@okta/vuepress-site/docs/guides/registration-inline-hook/main/index.md [hunk 8]
```diff
@@ -0,0 +1,155 @@
...
+
+To associate the Registration Inline Hook with your Profile Enrollment policy:
+
+1. In the Admin Console, go to **Security > Profile Enrollment**.
+
+1. Click the Pencil icon to edit the policy and associate it with your Registration Inline Hook.
+
+1. In **Enrollment Settings**, click the More Options icon and then select **Edit**. Select **Allowed** for **Sign-up** in the **For new users** section.
+
+1. Select your hook from the drop-down menu for **Use the following inline hook** under the options for **For new users** that you've set up an activated previously ("Guide Registration Hook Code"). The drop-down menu displays all the Registration Inline Hooks you have created.
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/guides/registration-inline-hook/main/index.md:88]:
```suggestion
1. Select your hook from the drop-down menu for **Use the following inline hook** under the options for **For new users** that you set up and activated previously ("Guide Registration Hook Code"). The drop-down menu displays all the Registration Inline Hooks that you created.
```

File: packages/@okta/vuepress-site/docs/guides/registration-inline-hook/main/index.md [hunk 9]
```diff
@@ -0,0 +1,155 @@
...
+
+1. Click **Save**.
+
+Your Registration Inline Hook is now configured for Profile Enrollment. You are now ready to preview and test the example.
+
+> **Note:** Only one Inline Hook can be associated with your Profile Enrollment policy at a time.
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/guides/registration-inline-hook/main/index.md:94]:
```suggestion
> **Note:** You can associate only one Inline Hook with your Profile Enrollment policy at a time.
```

File: packages/@okta/vuepress-site/docs/guides/registration-inline-hook/main/index.md [hunk 10]
```diff
@@ -0,0 +1,155 @@
...
+
+> **Note:** Self-service registration and Registration Inline Hooks are only supported with the [Okta Sign-In Widget](/code/javascript/okta_sign-in_widget/) version 2.9 or later.
+
+To enable the Registration Inline Hook on the self-service registration page:
+
+1. In the Admin Console, go to **Directory** > **Self-Service Registration**.
+
+1. Click **Edit**.
+
+1. From the **Extension** field drop-down menu, select the hook you've set up and activated previously ("Guide Registration Hook Code").
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/guides/registration-inline-hook/main/index.md:109]:
```suggestion
1. From the **Extension** drop-down menu, select the hook that you set up and activated previously ("Guide Registration Hook Code").
```

File: packages/@okta/vuepress-site/docs/guides/registration-inline-hook/main/index.md [hunk 11]
```diff
@@ -0,0 +1,155 @@
...
+To preview the Registration Inline Hook:
+
+1. In the Admin Console, go to **Workflow** > **Inline Hooks**.
+2. Select the Registration Inline Hook name (in this example, "Guide Registration Hook Code").
+3. Click the **Preview** tab.
+4. Select a user from your org in the first block titled "Configure Inline Hook request"; that is, a value for the  `data.userProfile` value.
+5. From the "Preview example Inline Hook request" block, click **Generate Request**.
+    You should see the user's request information in JSON format that is sent to the external service.
+6. From the "View service's response" block, click **View Response**.
+    You will see the response from your external service in JSON format, which either allows or denies the self-registration.
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/guides/registration-inline-hook/main/index.md:132]:
```suggestion
    You should see the response from your external service in JSON format, which either allows or denies the self-registration.
```

File: packages/@okta/vuepress-site/docs/guides/registration-inline-hook/main/index.md [hunk 12]
```diff
@@ -0,0 +1,155 @@
...
+
+### Test
+
+To run a test of your Registration Inline Hook, go to the Okta sign-in page for your Okta org, click the "Sign Up" link, and attempt to self-register.
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/guides/registration-inline-hook/main/index.md:136]:
```suggestion
To run a test of your Registration Inline Hook, go to the Okta sign-in page for your Okta org, click the **Sign Up** link, and attempt to self-register.
```

File: packages/@okta/vuepress-site/docs/guides/registration-inline-hook/main/index.md [hunk 13]
```diff
@@ -0,0 +1,155 @@
...
+In your Okta org, you can preview the request and response JSON right from the Admin Console. You can also test the code directly with self-registering users.
+
+### Preview
+
+To preview the Registration Inline Hook:
+
+1. In the Admin Console, go to **Workflow** > **Inline Hooks**.
+2. Select the Registration Inline Hook name (in this example, "Guide Registration Hook Code").
+3. Click the **Preview** tab.
+4. Select a user from your org in the first block titled "Configure Inline Hook request"; that is, a value for the  `data.userProfile` value.
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/guides/registration-inline-hook/main/index.md:128]: wouldn't this be `data.userProfile` **object**??

Also - there's an extra space in front of `data.userProfile`

File: packages/@okta/vuepress-site/docs/guides/registration-inline-hook/main/index.md [hunk 14]
```diff
@@ -0,0 +1,155 @@
...
+    You should see the user's request information in JSON format that is sent to the external service.
+6. From the "View service's response" block, click **View Response**.
+    You will see the response from your external service in JSON format, which either allows or denies the self-registration.
+
+### Test
+
+To run a test of your Registration Inline Hook, go to the Okta sign-in page for your Okta org, click the "Sign Up" link, and attempt to self-register.
+
+* If you use an allowable email domain, such as `john@example.com`, the user registration goes through.
+* If you use an incorrect email domain, the user registration is denied; review the error message, which displays the error summary from the external service code and is passed back to Okta.
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/guides/registration-inline-hook/main/index.md:139]: sentence is way too long with the semi-colon

```suggestion
* If you use an incorrect email domain, the user registration is denied. Review the error message, which displays the error summary from the external service code and is passed back to Okta.
```

**Review by susanharper-okta — APPROVED** [review]
couple things - approving not to hold you up

File: packages/@okta/vuepress-site/docs/guides/registration-inline-hook/main/index.md (hunk 2 above)
Comment (brianduffield-okta) [reply packages/@okta/vuepress-site/docs/guides/registration-inline-hook/main/index.md:155]: I simplified a bit to adhere to our style: 

For a complete description of this Inline Hook type, see[Registration Inline Hook](/docs/reference/registration-hook/).

File: packages/@okta/vuepress-site/docs/guides/registration-inline-hook/main/index.md (hunk 13 above)
Comment (brianduffield-okta) [reply packages/@okta/vuepress-site/docs/guides/registration-inline-hook/main/index.md:128]: Yes! Good catch. Thank you.

**Pull request merged**